File: Python/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        ans = ""
        if len(strs) == 0:
            return ans
        for i in range(len(strs[0])):
            for s in strs:
                if i >= len(s) or s[i] != strs[0][i]:
                    return ans
            ans += strs[0][i]
        return ans

File: C++/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        if(strs.size()==0)
            return ans;
        int len = strs[0].length();
        for(int i = 0;i<len;i++)
        {
            for(auto str: strs)
            {
                if(i>=str.length() || str[i] != strs[0][i])
                    return ans;
            }
            ans += strs[0][i];
        }
        return ans;
    }
};